mithril-aggregator/src/configuration.rs

```diff
@@ -262,19 +262,24 @@ impl Configuration {
     pub fn list_allowed_signed_entity_types_discriminants(
         &self,
     ) -> StdResult<Vec<SignedEntityTypeDiscriminants>> {
-        let mut signed_entity_types = Vec::new();
-        signed_entity_types.push(SignedEntityTypeDiscriminants::MithrilStakeDistribution);
-        signed_entity_types.push(SignedEntityTypeDiscriminants::CardanoImmutableFilesFull);
+        let default_signed_entity_types = vec![
+            SignedEntityTypeDiscriminants::MithrilStakeDistribution,
+            SignedEntityTypeDiscriminants::CardanoImmutableFilesFull,
+        ];
+
+        let mut all_signed_entity_types = default_signed_entity_types;
 
         let discriminant_names = self.signed_entity_types.clone().unwrap_or_default();
-        let mut signed_entity_types_appended = discriminant_names
+        for discriminant in discriminant_names
             .split(',')
             .filter_map(|name| SignedEntityTypeDiscriminants::from_str(name.trim()).ok())
-            .filter(|signed_entity_type| !signed_entity_types.contains(signed_entity_type))
-            .collect::<Vec<_>>();
-        signed_entity_types.append(&mut signed_entity_types_appended);
+        {
+            if !all_signed_entity_types.contains(&discriminant) {
+                all_signed_entity_types.push(discriminant)
+            }
+        }
```

Comment:
I would prefer to partially keep it like before with all filters expressed in the iterator instead of splitting the conditions between the iterator and and if.
```suggestion
            .filter(|entity_type| !all_signed_entity_types.contains(entity_type))
        {
            all_signed_entity_types.push(discriminant)
        }
```